src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            FruitTransaction.Operation operation = getOperationByCode(parts[0]);
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
```

Comment:
You should validate the length of 'parts' to ensure that the array has the expected number of elements before accessing them. This will prevent `ArrayIndexOutOfBoundsException` if the input data is malformed.